content/snippets/easy_to_change.md

```diff
@@ -0,0 +1,32 @@
+title: Easy to Test, Easy To Change
+tags: software engineering
+author: cdent
+created: 2024-02-23
+
+In conversation with my wife recently I mentioned that I've been doing software
+development for more than 40 years. I've learned, forgotten, and re-learned
+many things in that time.
+
+There are two goals I now accept as true and self evident but are neither
+readily nor universally implemented, especially in high pressure product
+development environments where short-term progress is traded against long-term
+success.
+
+Those two goals are more important than "providing the functionality we want".
+They are that code must be:
+
+1. easy to change
+2. easy to test
+
+Without these, maintaining the functionality we want (as that functionality
+evolves) becomes far too hard.
+
+And that functionality _will and must_ evolve. It must change, frequently. If
+we are to have any hope of doing that efficiently or effectively the change
+needs to be easy and it needs to be safe.
+
+Many software developers would accept these needs as true–in fact it is
+repeated as common lore in all the places you might look–but few have the
```

Comment:
```suggestion
Many software developers would accept these needs as true -- in fact it is
repeated as common lore in all the places you might look -- but few have the
```

Markdown takes care of this conversion (will become an en dash)

Reply:
well, since I want an em-dash, not at en-dash, what I've done is correct isn't it?

Reply:
fair (FCVObvsly) - FWIW, I've checked and `as true---in fact` works just the same